src/main/java/com/nha/abdm/wrapper/hip/HIPClient.java

```diff
@@ -89,4 +97,26 @@ public ResponseEntity<HealthInformationBundleResponse> healthInformationBundleRe
         .toEntity(HealthInformationBundleResponse.class)
         .block();
   }
+
+  public ResponseEntity<ProfileAcknowledgement> shareProfile(
+      ShareProfileRequest shareProfileRequest) {
+    return webClient
+        .post()
+        .uri(shareProfilePath)
+        .accept(MediaType.APPLICATION_JSON)
+        .body(BodyInserters.fromValue(shareProfileRequest))
+        .retrieve()
+        .toEntity(ProfileAcknowledgement.class)
+        .block();
+  }
+  public <T> ResponseEntity<RequestStatusResponse> fetchResponseFromHIP(String uri, T request) {
+    return webClient
+            .post()
+            .uri(uri)
+            .body(BodyInserters.fromValue(request))
+            .retrieve()
+            .toEntity(RequestStatusResponse.class)
```

Comment:
I dont think this should be RequestStatusResponse.. you can create a new class which reflects the attributes for this behavior.

Reply:
Since we only need acknowledgment, I am using that RequestStatusResponse.class, I can use a different class if needed.